src/Microsoft.AspNetCore.Razor.Evolution/DefaultRazorSourceDocument.cs

```diff
@@ -10,6 +10,7 @@ namespace Microsoft.AspNetCore.Razor.Evolution
     internal class DefaultRazorSourceDocument : RazorSourceDocument
     {
         private MemoryStream _stream;
+        private string _content;
 
         public DefaultRazorSourceDocument(MemoryStream stream, Encoding encoding, string filename)
```

Comment:
We can take a Roslyn approach and have different types of `RazorSourceDocument`s based on how they're constructed. Aka `LargeStringDocument`, `StringDocument`, `FileDocument`. Garbage names but you get the idea 😄